File: python/leetcode/P2419.py
class Solution:
	def longestSubarray(self, nums: list[int]) -> int:
		maxval = ans = currentStreak = 0;

		for num in nums:
			if num > maxval:
				maxval = num
				ans = currentStreak = 0

			if num == maxval:
				currentStreak += 1
			else:
				currentStreak = 0

			ans = max(ans, currentStreak)

		return ans

File: java/leetcode/P2419.java
public class P2419 {
	public int longestSubarray(int[] nums) {
		int maxval = 0;
		int ans = 0;
		int currentStreak = 0;

		for (int num : nums) {
			if (num > maxval) {
				maxval = num;
				ans = currentStreak = 0;
			}

			if (num == maxval) {
				currentStreak++;
			} else {
				currentStreak = 0;
			}

			ans = ans > currentStreak ? ans : currentStreak;
		}

		return ans;
	}
}
